cni/pkg/nodeagent/informers.go

```diff
@@ -83,6 +83,11 @@ func setupHandlers(ctx context.Context, kubeClient kube.Client, dataplane MeshDa
 }
 
 func (s *InformerHandlers) GetPodIfAmbient(podName, podNamespace string) (*corev1.Pod, error) {
+	// Both the CNI plugin and the informer check the K8S API to validate that the pod in
+	// question should be part of ambient, on "both ends".
+	// In the informer, wait for cache sync here to make sure we are fresh, to avoid issues
+	// where this cache is out of date with what the plugin might have seen when contacting the k8s API directly.
```

Comment:
This doesn't actually verify its up to date with k8s. It means it synced _once_ -- we still may be behind on updates.

Any reads from an informer without doing WaitForCacheSync are just broken because they may be done before the initial `List()` call so will always be entirely empty

Reply:
Ah so it makes no sense to invoke it more than once.

Ok, then I misunderstood what it was for.